glib-macros/src/lib.rs

```diff
@@ -468,6 +468,75 @@ pub fn enum_derive(input: TokenStream) -> TokenStream {
     gen.into()
 }
 
+/// Derive macro for register a rust enum in the glib type system as a dynamic
+/// type and derive the [`glib::Value`] traits.
+///
+/// An enum must be explicitly registered as a dynamic type when the system
+/// loads its implementation (see [`TypePlugin`] and [`TypeModule`].
+/// Therefore, whereas an enum can be registered only once as a static type,
+/// it can be registered several times as a dynamic type.
+///
+/// An enum registered as a dynamic type is never unregistered. The system
+/// calls [`TypePluginExt::unuse`] to unload its implementation. If the
+/// [`TypePlugin`] subclass is a [`TypeModule`], the enum registered as a
+/// dynamic type is marked as unloaded and must be registered again when the
+/// module is reloaded.
+///
+/// This macro provides two behaviors when registering an enum as a dynamic
+/// type:
+///
+/// By default an enum is registered as a dynamic type when the system loads
+/// its implementation (e.g. when the module is loaded):
+/// ```ignore
+/// use glib::prelude::*;
+/// use glib::subclass::prelude::*;
+///
+/// #[derive(Debug, Copy, Clone, PartialEq, Eq, glib::DynamicEnum)]
+/// #[enum_type(name = "MyEnum")]
+/// enum MyEnum {
+///     Val,
+///     #[enum_value(name = "My Val")]
+///     ValWithCustomName,
+///     #[enum_value(name = "My Other Val", nick = "other")]
+///     ValWithCustomNameAndNick,
+/// }
+/// ```
+///
+/// Optionally setting the macro attribute `lazy_registration` to `true`
+/// postpones registration on the first use (when `static_type()` is called for
+/// the first time), similarly to the [`macro@enum_derive`] macro:
+/// ```ignore
+/// #[derive(Debug, Copy, Clone, PartialEq, Eq, glib::DynamicEnum)]
+/// #[enum_type(name = "MyEnum", lazy_registration = true)]
+/// enum MyEnum {
+///     ...
+/// }
+/// ```
+///
+/// By default an enum is considered to be registered as a dynamic type within
+/// a [`TypeModule`] subclass. Optionally setting the macro attribute
```

Comment:
Maybe add a bit more details / example for what it means to be "considered to be registered as a dynamic type within a TypeModule" :)

Reply:
Thanks!